src/test/store/profile-view.test.js

```diff
@@ -835,6 +835,148 @@ describe('actions/ProfileView', function() {
+            messageSeqno: 1,
+            messageType: 'PContent::Msg_PreferenceUpdate',
+            side: 'child',
+            direction: 'receiving',
+            sync: false,
+          },
+        ],
+        ['c', 2, null],
+        [
+          'd',
+          3,
+          {
+            type: 'IPC',
+            startTime: 40,
+            endTime: 40,
+            otherPid: 9999,
+            messageSeqno: 2,
+            messageType: 'PContent::Msg_PreferenceUpdate',
+            side: 'parent',
+            direction: 'sending',
+            sync: false,
+          },
+        ],
+      ]);
+      const { dispatch, getState } = storeWithProfile(profile);
+
+      expect(
+        selectedThreadSelectors.getSearchFilteredMarkerIndexes(getState())
+      ).toHaveLength(4);
+      dispatch(ProfileView.changeMarkersSearchString('PContent'));
+
+      const getMarker = selectedThreadSelectors.getMarkerGetter(getState());
+      const markerIndexes = selectedThreadSelectors.getSearchFilteredMarkerIndexes(
+        getState()
+      );
+      expect(markerIndexes).toHaveLength(2);
+      expect(getMarker(markerIndexes[0]).name.includes('IPCIn')).toBeTruthy();
+      expect(getMarker(markerIndexes[1]).name.includes('IPCOut')).toBeTruthy();
+    });
+
+    it('filters the markers by other properties of a potential data payload', function() {
+      const profile = getProfileWithMarkers([
+        [
+          'a',
+          0,
+          {
+            type: 'tracing',
+            category: 'DOMEvent',
+            timeStamp: 1001,
+            interval: 'start',
+            eventType: 'mousedown',
+            phase: 1,
+          },
+        ],
+        [
+          'b',
+          1,
+          {
+            type: 'UserTiming',
+            startTime: 1002,
+            endTime: 1022,
+            name: 'mark-1',
+            entryType: 'mark',
+          },
+        ],
+        ['c', 2, null],
+        [
+          'd',
+          3,
+          {
+            type: 'UserTiming',
+            startTime: 1050,
+            endTime: 1100,
+            name: 'measure-1',
+            entryType: 'measure',
+          },
+        ],
+      ]);
+      const { dispatch, getState } = storeWithProfile(profile);
+
+      expect(
+        selectedThreadSelectors.getSearchFilteredMarkerIndexes(getState())
+      ).toHaveLength(4);
+      dispatch(ProfileView.changeMarkersSearchString('event, mark, measure'));
+
+      const getMarker = selectedThreadSelectors.getMarkerGetter(getState());
+      const markerIndexes = selectedThreadSelectors.getSearchFilteredMarkerIndexes(
+        getState()
+      );
+      expect(markerIndexes).toHaveLength(3);
+      expect(getMarker(markerIndexes[0]).name.includes('a')).toBeTruthy();
+      expect(getMarker(markerIndexes[1]).name.includes('b')).toBeTruthy();
+      expect(getMarker(markerIndexes[2]).name.includes('d')).toBeTruthy();
+    });
   });
```

Comment:
I see we're also missing tests around the marker's category, can you please add one?

Reply:
I've added an additional check for the category. There's a comment in the function we're testing that treats it as a property of a UserTiming marker. 

https://github.com/firefox-devtools/profiler/blob/e408b1845baa3bce381689a0d41ceff939007418/src/profile-logic/marker-data.js#L154-L157

Shouldn't it be a property of DOMEvents marker data?

Reply:
I think the comment here is incorrect, and was a bad copy/paste from the previous if block.